File: 🟡 Medium/1894 - Find the Student that Will Replace the Chalk/_1894FindtheStudentthatWillReplacetheChalk.js
/**
 * @param {number[]} chalk
 * @param {number} k
 * @return {number}
 */
var chalkReplacer = function (chalk, k) {
    let chalkSum = 0;

    for (let chalkPiece of chalk) {
        chalkSum += chalkPiece;
    }

    k %= chalkSum;
    let idx;
    for (idx = 0; idx < chalk.length; idx++) {
        if (chalk[idx] > k) {
            return idx;
        } else {
            k -= chalk[idx];
        }
    }

    return idx;
};

File: 🟡 Medium/1894 - Find the Student that Will Replace the Chalk/_1894FindtheStudentthatWillReplacetheChalk.c
int chalkReplacer(int* chalk, int chalkSize, int k) {
    long chalkSum = 0;

    for (int i = 0; i < chalkSize; i++) {
        chalkSum += chalk[i];
    }

    k %= chalkSum;
    int idx;
    for (idx = 0; idx < chalkSize; idx++) {
        if (chalk[idx] > k) {
            return idx;
        } else {
            k -= chalk[idx];
        }
    }

    return idx;
}